woocommerce/wc-admin
# Migrate the devdocs examples to Storybook CSF stories

Partially fixes  #5234 

This is phase 1 of the work to move to Storybook for documenting `@woocommerce/components`. In this phase our proprietary `devdocs` component format is removed in favour of the Storybook [CSF](https://storybook.js.org/docs/react/api/csf) API.

Because this is already a large changeset, I decided to separate out the removal of `devdocs` into a separate PR. I'm happy to hold off merging this until that PR is merged.

Some notes:

1. Most of our components come with a README currently. We should consider distilling some of the markdown in these READMEs into Storybook's MDX format later, but it would be a huge job. I think it should be updated piece by piece.

2. Not all the styles work perfectly. I have removed most of the issues with styles by importing the Wordpress wp-admin styles but a few anomalies remain. Some unfortunately are situations where we have some very broad styles that impact components when used in WooCommerce, these are tech debt that should be improved at some stage. I am thinking about raising a new issue for this.

3. If this PR is merged as is `devdocs` will stop working, so please keep a note of that, I don't want to merge this till the PR for removal of `devdocs` is done.

4. We're definitely not using Storybook to its fullest, a lot of components cannot currently have their props tweaked which is a lovely thing in Storybook. That would also be added to the tech debt to improve these component stories.


### Screenshot

![test](https://user-images.githubusercontent.com/1281828/95152356-6056ae80-07e9-11eb-8c82-201f355bdeb7.gif)


### Detailed test instructions:

1.  Run `npm i`
2. Run `npm run storybook`. The first run will take a while, it downloads the latest Wordpress wp-admin CSS
3. Check out storybook at the address specified in your terminal and have a look through the components.

### Changelog Note:

Dev: Migrate devdocs examples to Storybook stories
